### ISSUE TITLE
Provide href / onClick Property values as object

### ISSUE THREAD

**Issue by pickypg** [Paratron/hookrouter#24]
The current approach is to use `A` to do this directly.

```jsx
import { A } from 'hookrouter';

export const Link = () => (
  <A href="/">Root link</A>
);
```

This works great until you have a framework that wants to be given `href` / `onClick` properties for navigation. It would be nice if the internals of `A` were exported separately so that they could be reused, something like:

```js
export const getLinkProperties = (props) => {
  const onClick = (e) => {
    e.preventDefault();

    navigate(e.currentTarget.href);

    if (props.onClick) {
      props.onClick(e);
    }
  };

  const href = props.href.substr(0, 1) === '/'
               ? getBasepath() + props.href : props.href;

  return { href, onClick };
}
```

Then callers would be free to use it as:

```jsx
<MyLink whatever="x" {...getLinkProperties({ href: '/' })}>
  My link's text
</MyLink>
```

Then `A` becomes a simple class wrapper:

```jsx
const A = (props) => (
  <a {...props} {...getLinkProperties(props)} />
);
```

**Comment by FredyC** [comment]
I agree it would be great to have `useLink` so we can apply that logic to any component we like or even rename "href" to something else if needed.

**Comment by Paratron** [comment]
Released with `1.1.3` and documented, [here](https://github.com/Paratron/hookrouter/blob/master/src-docs/pages/en/03_navigation.md#creating-custom-link-components).
